climpred/bootstrap.py

```diff
@@ -1,10 +1,59 @@
 import numpy as np
 import xarray as xr
 
-from .prediction import compute_perfect_model, compute_persistence_pm
+from .metrics import POSITIVELY_ORIENTED_METRICS
+from .prediction import (compute_perfect_model, compute_persistence,
+                         compute_reference)
 from .stats import DPP, xr_varweighted_mean_period
 
 
+def resample_uninit_dple_from_lens(dple, lens):
+    """Resample uninitialized hindcast from historical members."""
+    # find range for bootstrapping
+    first_init = max(lens.time.min().values, dple.init.min().values)
+    last_init = min(lens.time.max().values - dple.lead.size,
+                    dple.init.max().values)
+    dple = dple.sel(init=slice(first_init, last_init))
+
+    uninit_dple = []
+    for init in dple.init.values:
+        random_members = np.random.choice(
+            lens.member.values, dple.member.size)
+        # take random uninitialized members from lens at init forcing
+        # (Goddard allows 5 year forcing range here)
+        uninit_at_one_init_year = lens.sel(
+            time=slice(init + 1, init + dple.lead.size)).sel(
+                member=random_members).rename({'time': 'lead'})
+        uninit_at_one_init_year['lead'] = np.arange(
+            1, 1 + uninit_at_one_init_year.lead.size)
+        uninit_at_one_init_year['member'] = np.arange(1,
+                                                      1 + len(random_members))
+        uninit_dple.append(uninit_at_one_init_year)
+    uninit_dple = xr.concat(uninit_dple, 'init')
+    uninit_dple['init'] = dple.init.values
+    return uninit_dple
+
+
+def _distribution_to_ci(ds, ci_low, ci_high, dim='bootstrap'):
+    """Get confidence intervals from bootstrapped distribution."""
+    try:  # incase of lazy results compute
+        if len(ds.chunks) >= 1:
+            ds = ds.compute()
+    except:
+        pass
+    ds_ci = ds.quantile(q=[ci_low, ci_high], dim=dim)
+    return ds_ci
+
+
+def _pvalue_from_distributions(simple_fct, init, metric='pearson_r'):
+    """Get probability that skill of simple_fct is larger than
+    init skill."""
+    pv = ((simple_fct - init) > 0).sum('bootstrap') / init.bootstrap.size
+    if metric not in POSITIVELY_ORIENTED_METRICS:
+        pv = 1 - pv
+    return pv
+
+
 def pseudo_ens(ds, control):
```

Comment:
How is this different from `resample_uninit_dple_from_lens` now? The goal for both is to take something with a `time` dimension and make it have `time`, `init`, `lead`? Except I guess `resample_uninit_dple_from_lens` expects multiple members?

It might be nice to have these functions have similar names to show that they are both resampling. `resample_from_control` and `resample_from_uninit_ensemble` maybe?